ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                        .nNumberOfDecimals = 0;
+                    break;
+
+                case OFTInteger64:
+                case OFTInteger64List:
+                    phMiraMonLayer->pLayerDB->pFields[iField].bIs64BitInteger =
+                        TRUE;
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Numeric;
+                    phMiraMonLayer->pLayerDB->pFields[iField]
+                        .nNumberOfDecimals = 0;
+                    break;
+
+                case OFTReal:
+                case OFTRealList:
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Numeric;
+                    phMiraMonLayer->pLayerDB->pFields[iField]
+                        .nNumberOfDecimals =
+                        poFeatureDefn->GetFieldDefn(iField)->GetPrecision();
+                    break;
+
+                case OFTBinary:
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Logic;
+                    break;
+
+                case OFTDate:
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Data;
+                    break;
+
+                case OFTTime:
+                case OFTDateTime:
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Character;
+                    break;
+
+                case OFTString:
+                case OFTStringList:
+                default:
+                    phMiraMonLayer->pLayerDB->pFields[iField].eFieldType =
+                        MM_Character;
+                    break;
+            }
+            if (poFeatureDefn->GetFieldDefn(iField)->GetType() == OFTDate)
+                phMiraMonLayer->pLayerDB->pFields[iField].nFieldSize = 8;
+            else
+            {
+                if (poFeatureDefn->GetFieldDefn(iField)->GetPrecision() == 0)
+                {
+                    phMiraMonLayer->pLayerDB->pFields[iField].nFieldSize =
+                        poFeatureDefn->GetFieldDefn(iField)->GetWidth();
+                    if (phMiraMonLayer->pLayerDB->pFields[iField].nFieldSize ==
+                        0)
+                        phMiraMonLayer->pLayerDB->pFields[iField].nFieldSize =
+                            1;
+                }
+                else
+                {
+                    // One more space for the "."
+                    phMiraMonLayer->pLayerDB->pFields[iField].nFieldSize =
+                        (unsigned int)(poFeatureDefn->GetFieldDefn(iField)
+                                           ->GetWidth() +
+                                       1);
+                }
+            }
+
+            if (poFeatureDefn->GetFieldDefn(iField)->GetNameRef())
+            {
+                // Interlis 1 encoding is ISO 8859-1 (Latin1) -> Recode from UTF-8
```

Comment:
"Interlis 1" :-) ? must be a copy&paste left over

Reply:
Done